src/hooks/useERC20.tsx

```diff
@@ -0,0 +1,106 @@
+import React from "react"
+import { ERC20 } from "../contracts"
+import ERC20ABI from "../abi/ERC20.json"
+import { useAccount, useContract, useProvider, useSigner } from "wagmi"
+import { BigNumber } from "ethers"
+
+/**
+ * Object containing used ERC20 tokens.
+ *
+ * Adding a new token will update type definitions as well.
+ */
+const TokenContracts = {
+  USDC: process.env.REACT_APP_USDC_ADDRESS as string,
+}
+
+type AvailableERC20Tokens = keyof typeof TokenContracts
+
+/**
+ * Hook for interacting with an ERC20 token contract
+ * @param token ERC20 Token
+ * @param contractAddress Contract address for an ERC20 token
+ */
+const useERC20 = (token?: AvailableERC20Tokens, contractAddress?: string) => {
```

Comment:
great job creating this hook. I can include `SHER` later on.

Reply:
Thank you! Not sure yet if this is the best approach, as it might create multiple repeated requests if used in multiple components at the same time (I think nested components make it worse, but I'm not sure). I was thinking about moving to a context with provider pattern, but I wanted to ask for your opinion as well.

Reply:
ah yes, I see what you mean. Context and Provider is a pattern I like, as long as we don't end up with a gigantic pile of providers in the hierarchy.

I've been there a re-renders might be a nightmare.

Reply:
I agree, falling in that trap would end up causing just as many problems as not doing it.

Should we reconsider this problem later on, if we encounter issues with current implementation?

Reply:
yep, happy to keep it this way for now 👍🏽